[Ivan Theng Wen Rong] iP

ivanthengwr opened the pull request.

ivanthengwr committed:
Add initial skeletal version of the Flash

ivanthengwr committed:
Add Greet, Echo, Exit on skeletal of Flash

ivanthengwr committed:
Add ability to store and list user input text

ivanthengwr committed:
Add Mark/Unmark as Done

ivanthengwr committed:
Add Task class to represent tasks

ivanthengwr committed:
Update code to comply with coding standard

ivanthengwr committed:
Update code to comply with coding standard

ivanthengwr committed:
Update code to comply with coding standard

ivanthengwr committed:
Add and update new classes and functionality

Created new classes such as ToDo, Deadline and Event to
make known the different types of tasks.

A superclass Task is inherited by the child classes to
allow inheritance of pre-existing instances to make it
more neat and easy to comprehand.

To carry out a Text Ui Testing followed by reviewing code
to aline with code quality standards.

ivanthengwr committed:
Update new expected and input text-ui-test

ivanthengwr committed:
Update code to improve Code Quality

Zhai-Yuxin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+
 public class Duke {
+    /** Array of assignments that is limited to 100 tasks */
+    private static final Task[] ASSIGNMENTS = new Task[100];
```
ASSIGNMENTS is not a constant so you may want to make it to be not capitalized

redders7 replied:
Consider naming in PascalCase (only first letter capitalised)

Zhai-Yuxin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+
+        System.out.println("\t Roger that. I've added this task:");
+
+        if(isToDo) {
```
you may want to put spacing after if

redders7 replied:
Consider adding a space after if

Zhai-Yuxin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+            System.out.println("\t " + numberOrder + ".["
+                    + ASSIGNMENTS[i].getStatusOfTypeTask() + "]["
+                    + ASSIGNMENTS[i].getStatusOfDone() + "] "
+                    + ASSIGNMENTS[i].displayTypeTaskDetails());
```
really nice formatting that make the printing code very easy to read

Zhai-Yuxin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+     * @param isMark boolean true or false to check if the task has been marked.
+     */
+    public static void printMarkOrUnmarkTask(int markIndex, boolean isMark) {
+        if(isMark) {
```
Same point as above, may want to add a spacing after if
there is similar places below too

Zhai-Yuxin reviewed (commented):
overall good syntax, but only minor spacing issue here and there
good job!

redders7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+    /**
+     * Returns the indexTask that is .
+     *
+     * @param assignment assignment that is based on the type of task.
```
Consider rephrasing comment for the assignement @param

redders7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+     * @return error which is a boolean that tell us if there is an error or not.
+     */
+    public static boolean hasErrorMarkTask(int markIndex, String lineDivider) {
+        boolean error = false;
```
Consider renaming to sound like a boolean (isError)

redders7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+     * @return error which is a boolean that tell us if there is an error or not.
+     */
+    public static boolean hasErrorUnMarkTask(int unMarkIndex, String lineDivider){
+        boolean error = false;
```
Consider renaming to sound like a boolean (isError)

redders7 reviewed (commented):
Everything looks good so far!

ishitamandal06 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+public class Deadline extends Task {
+    /** To check if the task is a deadline task */
+    protected boolean isDeadlineCheck;
```
Hi! I think adding the word 'checked' makes it sound like a function instead of a boolean variable. 
Maybe you could change it to is Deadline?

ishitamandal06 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+        addTask(new Event(splitUserInputs[1]));
+        ASSIGNMENTS[indexTask].markTypeTask();
+        System.out.println("\t   [" + ASSIGNMENTS[indexTask].getStatusOfTypeTask() + "]" + "[ ] "
+                + ASSIGNMENTS[indexTask].displayTypeTaskDetails());
```
It becomes hard to understand what exactly is going on in this print statement because there are functions being executed inside the print statement itself. Maybe you could seperately execute the functions, store them in variables, and then print the variables? It would definitely make it easier to understand what is going on. :)

ivanthengwr committed:
Update code to improve Code Quality

ivanthengwr committed:
Add user input error handling

ivanthengwr committed:
Improve Code Quality of markOrUnmark feature

ivanthengwr committed:
Add error handling for all types of tasks

ivanthengwr committed:
Add validation to markOrUnmarkTask

A new class have been added called DukeException. This class is used
to track for exceptions that are specific to Duke.java program.

Streamline the validation for typeOfTask, in coporated extra validation
to check and if it user input spaces instead of leaving it blank.

To address the user input for taskDetails next and validate to see if
user have typed in the event detail correctly or have left blank.
If time permit, else leave validation for wk6 ip

xseh commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String getStatusOfTypeTask() {
+        //mark Task with Deadline with "D"
```
Please consider removing the comment to avoid repeating the obvious,

xseh commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+    @Override
+    public String getStatusOfTypeTask() {
+        //mark Task with Deadline with "D"
+        return (isDeadline ? "D" : " ");
```
Avoid complicated expressions, can consider breaking down the logic into multiple statements.

xseh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,236 @@
+     * @param splitUserInputs array of string that have been split into two.
+     */
+    public static void sortTypeOfTask(String userInput, String[] splitUserInputs) {
+        boolean isToDo = splitUserInputs[0].equals("todo");
+        boolean isDeadlines = splitUserInputs[0].equals("deadline");
+        boolean isEvent = splitUserInputs[0].equals("event");
```
1. Avoid using magic numbers. Consider putting the numbers into a descriptive variable name to make the code easier to understand. For example, `0` can be put inito a variable `COMMAND_INDEX`. 

1. Consider abstracting the logic into methods (i.e. SLAP more) to make the code more understandable.

xseh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,236 @@
+            String[] splitUserInputs = userInput.split(" ", 2);
+            isBye = splitUserInputs[0].equals("bye");
+
+            boolean isUserInputData = !splitUserInputs[0].equals("bye")
+                    && !splitUserInputs[0].equals("list")
+                    && !splitUserInputs[0].equals("mark")
+                    && !splitUserInputs[0].equals("unmark");
```
Avoid complicated expressions. Perhaps you can consider splitting the logic up into multiple statements and abstract similar methods (SLAP) to avoid repeating the code. E.g. a function `isUserInputValid(string userInput)` can be used to check for valid user input.

xseh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,236 @@
+                System.out.println("\t Bye. Hope to see you again soon!");
+            }
+            System.out.println("\t" + lineDivider);
+        } while(!isBye);
```
This code violates the coding standard. Consider adding spaces between `while` and `(` to comply with coding standards.

xseh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,236 @@
+                + "] " + assignments[markIndex].description);
+    }
+
     public static void main(String[] args) {
```
Please consider avoiding long methods with more abstraction. 

As stated on the module website, 
> Be wary when a method is longer than the computer screen, and take corrective action when it goes beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.

xseh reviewed (commented):
Overall good effort! I like that you took initiative with the javadoc comments to make the code easily understood. However, please take note of coding standards (especially the spaces), SLAP more and avoid commenting the obvious as well as complicated expressions.

ivanthengwr committed:
Update to improve Code

Edited the code to improve both coding standards and code quality.
Noted the feedback from the pull request and had made the necessary
changes.

Will be commiting this and merging to master branch before pushing.

ivanthengwr committed:
Merge branch 'branch-Level-5'

ivanthengwr committed:
Add a few packages to the project

More specifically, packages that consist of duke as overall and sub
packages such as duke,command and duke.task.

ivanthengwr committed:
Merge branch 'A-Packages'

ivanthengwr committed:
Add delete function to Duke

Added delete functionality to duke and also updated duke to improve Code
Quality. Made use of abstract on Task to simply code.

Will be handling some errors for delete in next commit.

ivanthengwr committed:
Add error handling for delete

Added error handling when user does not specify what to delete or
specify something that is unknown.

Updated the print list to include a message that notify the user if
there are no task in the list at present.

Will create tag for Level-6 and A-Collections and will create new
breanch-Level-7 in master for Level-7 task.

ivanthengwr committed:
Add a create DataFile class to create duke.txt

Add a DataFile class to class a duke.txt if it does not exist.
Else, the bot will notify the User that it already existed.`

ivanthengwr committed:
Add save to file, update file, read file

Added two new classes which is DataFile and EditDataFile to read, write
and update the file each time the user updates the task list.`

ivanthengwr committed:
Delete man file as it is not needed

ivanthengwr committed:
Update EXPECTED.txt to include latest output.

ivanthengwr committed:
Merge branch 'branch-Level-6'

ivanthengwr committed:
Merge branch 'branch-Level-7'

Resolved conflict and edit the source code.
To improve code quality and coding standards

ivanthengwr committed:
Update code to improve quality and standard.

ivanthengwr committed:
Update dataFile "./data/data.txt" relative path

ivanthengwr committed:
Add more to OOP, improve code quality

Added a few new classes to make functions more distinct.

ivanthengwr committed:
Add in Dates and Times

ivanthengwr committed:
Add in find function and improve code quality

Added a findCommand class to enable find function for the user. Improved
code quality and standards.

ivanthengwr committed:
Update a function name

ivanthengwr committed:
Add data/tasks/txt back to dir

ivanthengwr committed:
Add JavaDoc comments to the code

Added comments to the code and also changed a few function names

ivanthengwr committed:
Merge pull request #1 from ivanthengwr/branch-Level-8

Add in Dates and Times

ivanthengwr committed:
Merge branch 'master' into branch-Level-9

ivanthengwr committed:
Merge pull request #2 from ivanthengwr/branch-Level-9

Branch level 9

ivanthengwr committed:
Update function that is resolved wrongly during merge

ivanthengwr committed:
Merge branch 'master' into branch-A-JavaDoc

ivanthengwr committed:
Merge pull request #3 from ivanthengwr/branch-A-JavaDoc

Add JavaDoc comments to the code

ivanthengwr committed:
Update readme file by writing up a user guide

ivanthengwr committed:
Update welcome display
